crates/bindings-macro/src/lib.rs

```diff
@@ -436,13 +435,80 @@ struct Column<'a> {
     attr: ColumnAttribute,
 }
 
-fn spacetimedb_table(item: TokenStream, public: Option<Span>) -> syn::Result<TokenStream> {
+fn spacetimedb_table(item: TokenStream, public: Option<Span>, scheduled: Option<Ident>) -> syn::Result<TokenStream> {
     let public = public.map(|span| quote_spanned!(span => #[sats(public)]));
 
+    let output = if let Some(reducer) = scheduled {
+        schedule_table(item, reducer, public)?
+    } else {
+        quote! {
+            #[derive(spacetimedb::TableType)]
+            #public
+            #item
+        }
+    };
+    Ok(output)
+}
+
+fn schedule_table(item: TokenStream, reducer: Ident, public: Option<TokenStream>) -> syn::Result<TokenStream> {
+    let reducer_name = reducer.to_string();
+    let mut modified_item = syn::parse2::<DeriveInput>(item)?;
+    let type_check = reducer_type_check(&modified_item, &reducer)?;
+    add_scheduled_fields(&mut modified_item)?;
+    // codegen call all the `__preinit`s to add it to module Typespace
+    // codegen does not try to replace ref types with their product types.
+    // hence known types get registered as ref types.
+    let ref_register = quote! {
+        const _: () = {
+            #[export_name = "__preinit__20_register_describer_ScheduleAt"]
```

Comment:
This means that there can only be a single scheduled reducer per stdb module.

What would speak against adding "known types" to the typespace by default? 
`__describe_module__()` stands out as a place to do that.

cc @Centril @gefjon

Reply:
I think the problem is because `ScheduleAt` ref regestering is happening for all scheduled tables.
On anoher note, code generation is working fine without doing this as all. May be, `make_type` of `ScheduleAt` getting called early enough through some way. Still checking.

Reply:
It is because `ScheduleAt` is always a field of a struct, and parents `preinit` register describer defined for parent calls it `make_type`.

Reply:
Ok, so we don't actually need this preinit at all?

Reply:
Yes, I removed it testing generating and publishing module. Even by raw printing module typespace with and without this `preinit` they are exactly same.